pennylane/_qubit_device.py

```diff
@@ -283,6 +283,17 @@ def statistics(self, observables):
             elif obs.return_type is Probability:
                 results.append(self.probability(wires=obs.wires))
 
+            elif obs.return_type is State:
+                try:
+                    state = self.state
+                except AttributeError:
+                    state = None
+
+                if state is None:
```

Comment:
Though I'm not really well-versed in this things, I *think* the pythonic way to do this (essentially intercepting an exception and changing the message) is to `raise ... from ...`

Reply:
Also, this is something that a device can explicitly register whether it supports (in the device._capabilities` dictionary

Reply:
Good point. I still tend to prefer EAFP in cases like this (try and get the state, pass if there is an error). I worry about maintenance with the `device._capabilities` dictionary --- with EAFP, we can 'activate' new plugin features directly from core. Especially since the (current) device API explicitly raises a `NotImplementedError` if `state()` is not defined.

More complicated interconnection (such as passthru) should definitely use the capabilities attribute.

Reply:
Linked conversation https://github.com/PennyLaneAI/pennylane/pull/789/files#r486622239.